crates/bevy_ecs/src/schedule/mod.rs

```diff
@@ -206,6 +206,13 @@ impl Schedule {
         // of the game. Closures inherit generic parameters from their enclosing function.
         #[cold]
         fn stage_not_found(stage_label: &dyn Debug) -> ! {
+            if ["PreStartup", "Startup", "PostStartup"]
+                .contains(&format!("{:?}", stage_label).as_str())
```

Comment:
I don't think comparing on strings is a good idea, and I doubt it will be updated if for whatever reason we change the name of the startup stages. We could compare based on type id, but as `StartupStage` lives in `bevy_app` it's not available in `bevy_ecs`. Maybe we could move this enum over?

Reply:
I had the same thought but wasn't sure if moving the enum over was on the table. We could also add the check to `App::add_system_to_stage` and `App::add_system_set_to_stage` since startup seems to be an App concept as opposed to a bevy_ecs concept.

Reply:
> We could also add the check to App::add_system_to_stage and App::add_system_set_to_stage since startup seems to be an App concept as opposed to a bevy_ecs concept.

This makes the most sense to me.

Reply:
Agreed. I'm honestly not a big fan of adding special cases at all as it feels pretty hackey. But if we're going to do it, I think it should happen in `App`.